rasa/core/actions/action.py

```diff
@@ -331,6 +335,53 @@ def __init__(self) -> None:
         return evts + [Restarted()]
 
 
+class ActionSessionStart(Action):
+    """Applies a conversation session start.
+
+    Takes all `SlotSet` events from the previous session and applies them to the new
+    session.
+    """
+
+    def name(self) -> Text:
+        return ACTION_SESSION_START_NAME
+
+    @staticmethod
+    def _slot_set_events_from_tracker(
+        tracker: "DialogueStateTracker",
+    ) -> Generator["SlotSet", None, None]:
+        """Fetch SlotSet events from tracker and carry over key, value and metadata."""
+
+        from rasa.core.events import SlotSet
+
+        # use generator so the timestamps are greater than that of the returned
+        # SessionStarted event in the run() call
+        return (
+            SlotSet(key=event.key, value=event.value, metadata=event.metadata)
+            for event in tracker.events
```

Comment:
will this not also include any slotset events that come before a restart or the previous sessionstart?